src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogViewModel.java

```diff
@@ -0,0 +1,324 @@
+import org.fxmisc.easybind.EasyBind;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SharedDatabaseLoginDialogViewModel extends AbstractViewModel {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SharedDatabaseLoginDialogViewModel.class);
+
+    private final ListProperty<DBMSType> allDBMSTypes = new SimpleListProperty<>(FXCollections.observableArrayList(DBMSType.values()));
+    private final ObjectProperty<DBMSType> selectedDBMSType = new SimpleObjectProperty<>(allDBMSTypes.get(0));
+
+    private final StringProperty db = new SimpleStringProperty("");
+    private final StringProperty host = new SimpleStringProperty("");
+    private final StringProperty port = new SimpleStringProperty("");
+    private final StringProperty user = new SimpleStringProperty("");
+    private final StringProperty password = new SimpleStringProperty("");
+    private final StringProperty folder = new SimpleStringProperty("");
+    private final BooleanProperty autosave = new SimpleBooleanProperty();
+    private final BooleanProperty rememberPassword = new SimpleBooleanProperty();
+    private final BooleanProperty loading = new SimpleBooleanProperty();
+
+    private final JabRefFrame frame;
+    private final DialogService dialogService;
+    private final SharedDatabasePreferences prefs = new SharedDatabasePreferences();
+
+    private final Validator dbvalidator;
+    private final Validator hostvalidator;
+    private final Validator portValidator;
+    private final Validator uservalidator;
+    private final Validator folderValidator;
+    private final CompositeValidator formValidator;
+
+    private DBMSConnectionProperties connectionProperties;
+
+    public SharedDatabaseLoginDialogViewModel(JabRefFrame frame, DialogService dialogService) {
+        this.frame = frame;
+        this.dialogService = dialogService;
+
+        EasyBind.subscribe(selectedDBMSType, selected -> {
+            port.setValue(Integer.toString(selected.getDefaultPort()));
+        });
+
+        Predicate<String> predicate = input -> (input != null) && !input.trim().isEmpty();
+
+        dbvalidator = new FunctionBasedValidator<>(db, predicate, ValidationMessage.error(Localization.lang("Required field \"%0\" is empty.", Localization.lang("Library"))));
+        hostvalidator = new FunctionBasedValidator<>(host, predicate, ValidationMessage.error(Localization.lang("Required field \"%0\" is empty.", Localization.lang("Port"))));
+        portValidator = new FunctionBasedValidator<>(port, predicate, ValidationMessage.error(Localization.lang("Required field \"%0\" is empty.", Localization.lang("Host"))));
+        uservalidator = new FunctionBasedValidator<>(user, predicate, ValidationMessage.error(Localization.lang("Required field \"%0\" is empty.", Localization.lang("User"))));
+        folderValidator = new FunctionBasedValidator<>(folder, predicate, ValidationMessage.error(Localization.lang("Please enter a valid file path.")));
```

Comment:
I think we should use the nio-path methods to check if the provided path is actually valid.